cql/DashboardLibrary.cql

```diff
@@ -227,33 +237,69 @@ define function HpvInterpretation(conclusions List<System.Concept>):
  * @param interpretations - a list of tuples
  * @returns System.String the text to be used as input to the risk tables
  */
-define function HighestRankedInterpretation(interpretations List<Tuple { text System.String, rank System.Integer }>):
+define function HighestRankedInterpretation(interpretations List<Tuple { riskTableInput System.String, label System.String, rank System.Integer, grade System.Integer }>):
   Last(
     (
       (interpretations) I sort by rank
     ) S
-    return S.text
+    return S.riskTableInput
   )
 
 /**
+ * Set a flag if a result should be highlighted
+ * Takes the output from Interpretation() and returns the grade of the
+ * interpretation with the highest rank. This allows more specific conclusions to
+ * be highlighted in user interfaces.
+ * @param interpretations - a list of tuples
+ * @returns System.Integer the value used to determine the grade (class) of result
+ */
+define function InterpretedResultLabel(interpretations List<Tuple { riskTableInput System.String, label System.String, rank System.Integer, grade System.Integer }>):
+  Last(
+    (
+      (interpretations) I sort by rank
+    ) S
+    return S.label
+  )
+  
+/**
+ * Return the label to be used in user interfaces for an interpreted result.
+ * @param interpretations - a list of tuples
+ * @returns System.Integer the value used to determine the grade (class) of result
+ */
+define function InterpretedResultGrade(interpretations List<Tuple { riskTableInput System.String, label System.String, rank System.Integer, grade System.Integer }>):
+  Last(
+    (
+      (interpretations) I sort by rank
+    ) S
+    return S.grade
+  )
```

Comment:
Commented descriptions of two functions you added are swapped.

Reply:
These will be removed by using one HighestRankedInterpretation function